### ISSUE TITLE
Clarifying Additional Deps and Shared Store usage in troubleshooting doc

### ISSUE THREAD

**Issue by nrcventura** [open-telemetry/opentelemetry-dotnet-instrumentation#624]
This is a follow up for https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/623#issuecomment-1105775020.

The section below is hard to understand: 

https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/0683f38542d85f671e6a174f2498665a7cf0f21b/docs/troubleshooting.md?plain=1#L22-L25

We should clarify how to set the additional depends and shared store environment variables, as well as providing more information about [how the shared store affects probing paths](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-probing.md#probing-paths).

**Comment by RassK** [comment]
I'm still leaning towards auto setting these variables via native profiler. Seems less and less probable that auto instrumentation works without native profiler.

**Comment by nrcventura** [comment]
> I'm still leaning towards auto setting these variables via native profiler. Seems less and less probable that auto instrumentation works without native profiler.

I'm torn on this idea for a couple of reasons:

1. When we enable the profiler so that we can leverage byte-code injection, it results in disabling runtime performance improvements for newer versions of .NET. I have seen this as a concern from a subset of customers. If they do not need any of the instrumentation libraries that leverage byte-code injection, they should be able to experience better performance.
2. Some instrumentation libraries ran into dependency loading problems when we tried to bootstrap everything via byte-code injection. We technically could just set environment variables while the profiler is initializing. I think the downside to that approach is that it can make our setup instructions more complicated.

One thing that @pjanotti and I have talked about is conditionally enabling the profiler based on what instrumentations are needed so that we only take the performance hit when it's really necessary. We've also talked about experimenting with an API available from managed code that can trigger enabling the profiler. Both of these approaches would allow us to simplify our setup instructions and steps.
